enhancements/shared-and-guaranteed-cpu-pools.md

```diff
@@ -0,0 +1,604 @@
+on the node based on the shared/isolated CPUSets specified in the config file.
+7. Something schedules:
+   * a pod with the `target.workload.openshift.io/management` annotation in a namespace 
+with the `workload.openshift.io/allowed` management annotation. The admission hook ignores this pod as it will
+be handled by the managementcpusoverride admission hook.
+   * a pod with Burstable or BestEffort QoS. The admission hook modifies the pod,
+adding `openshift.io/shared-cpus` requests/limits for each container, matching the CPU requests.
+   * a pod with Guaranteed QoS. The admission hook modifies the pod as follows:
+     * for any containers with whole CPU requests/limits it adds `openshift.io/guaranteed-cpus` requests/limits
+     * for any containers with fractional CPU requests/limits it adds `openshift.io/shared-cpus` requests/limits
+8. The scheduler sees the new pod and finds available `openshift.io/shared-cpus` and/or 
+`openshift.io/guaranteed-cpus` resources on a node. The scheduler places the pod on the node.
+9. Kubelet processes the pod as usual, but when a guaranteed QoS container using whole CPUs is created, it 
+allocates CPUs from the `guaranteedCpuSet` instead of the `defaultCpuSet` and sets the CPU affinity to match.
+10. Repeat steps 7-9 until all pods are running.
+
+### API Extensions
+
+A new admission hook in the Kubernetes API Server within OpenShift will mutate pods when they are created
+to add `openshift.io/guaranteed-cpus` and/or `openshift.io/shared-cpus` requests/limits as described above.
+
+Note that the existing [Management Workload Partitioning](https://github.com/openshift/enhancements/blob/master/enhancements/workload-partitioning/management-workload-partitioning.md) feature will be a dependency for this
+feature. This will ensure that the API Server (and all other platform pods) with the 
+`target.workload.openshift.io/management` annotation will be placed on the reserved CPU set, even if they are
+started before the new admission hook is running.
+
+### Implementation Details/Notes/Constraints
+
+#### Cluster Wide Feature Scope
+
+One constraint of this approach is that the feature scope is cluster wide. When the feature gate is
+enabled, the new admission hook will mutate all incoming pods, adding requests/limits for the new
+extended resources. This means that all nodes in the cluster must have a Performance Profile that
+includes shared CPUs - without that, no pods would be scheduled on that node. Note that the isolated
+CPUs partition would be optional and would only be required on nodes that were going to run
+pods that had containers with whole CPU requests/limits.
+
+#### cgroup v1 vs v2 considerations
+
+This feature is cgroup version agnostic - it does not require any changes to cgroup configuration for
+pods or containers. The feature will work with either cgroup v1 or v2, without modification to the
+code introduced by this feature.
```

Comment:
Are there any additional difficulties introduced due to how load balancing is disabled in v2 now? See #1396 comments for how the cgroup heirarchy will need to be adjusted for that.

Reply:
No. This feature is agnostic to the cgroup version.
For containers requesting guaranteed cpus instead of allocating cpus from the defaultCpuset it is allocating cpus from a subset of the defaultCpuset. The runtime will just be given a set of cpus to be assigned to the container and would not impact how load balancing is being disabled.

Reply:
I updated this with some of the details from Brent's comment.